### PR TITLE
Add ops.is_tensor to the public API

### PR DESCRIPTION
**Pull request by grasskin**

### CODE REVIEW

File: keras/ops/core_test.py [hunk 1]
```diff
@@ -411,3 +411,9 @@ def fn(elems):
         self.assertAllClose(
             backend.convert_to_numpy(output), 2 * np.ones((2, 3))
         )
+
+    def test_is_tensor(self):
+        np_x = np.array([[1, 2, 3], [3, 2, 1]])
+        x = backend.convert_to_tensor(np_x)
+        self.assertFalse(ops.is_tensor(np_x))
+        self.assertTrue(ops.is_tensor(x))
```
Comment (tirthasheshpatel) [inline keras/ops/core_test.py:419]:
```suggestion
        if backend.backend() != "numpy":
            self.assertFalse(ops.is_tensor(np_x))
        self.assertTrue(ops.is_tensor(x))
        self.assertFalse(ops.is_tensor([1, 2, 3]))
```

**Review by tirthasheshpatel — APPROVED** [review]
LGTM! Thanks @grasskin!

File: keras/ops/core.py [hunk 2]
```diff
@@ -629,3 +630,19 @@ def vectorized_map(function, elements)
     a single list of tensor arguments.
     """
     return backend.core.vectorized_map(function, elements)
+
+
+@keras_export("keras.ops.is_tensor")
+def is_tensor(x):
+    """Check whether the given object is a tensor.
+
+    Note: This checks for backend specific tensors so passing a tensorflow
```
Comment (fchollet) [inline keras/ops/core.py:639]: TensorFlow

File: keras/ops/core.py [hunk 3]
```diff
@@ -629,3 +630,19 @@ def vectorized_map(function, elements)
...
+    tensor would return False if your backend is pytorch or jax.
```
Comment (fchollet) [inline keras/ops/core.py:640]: `False` (backticks)

PyTorch
JAX

File: keras/ops/core.py [hunk 4]
```diff
@@ -629,3 +630,19 @@ def vectorized_map(function, elements)
...
+
+    Args:
+        x: A variable.
+
+    Returns:
+        True if x is a tensor, otherwise False.
```
Comment (fchollet) [inline keras/ops/core.py:646]: `True`, `x`, `False` (backticks)

**Review by fchollet — COMMENTED** [review]
Thanks for the PR

File: keras/ops/core.py (hunk 3 above)
Comment (grasskin) [reply keras/ops/core.py]: TIL JAX used to stand for ""Just After eXecution" https://github.com/google/jax/discussions/9019

**Comment by divyashreepathihalli** [comment]
Thanks @grasskin for adding this!

**Review by divyashreepathihalli — APPROVED** [review]

**Review by fchollet — APPROVED** [review]

**Pull request merged**